src/fluree/db/json_ld/shacl.cljc

```diff
@@ -485,13 +486,13 @@
 
 (defn validate-shape
   "Check to see if each property shape is valid, then check node shape constraints."
-  [db {:keys [property] :as shape} s-flakes pid->p-flakes]
+  [db {:keys [property validated-properties] :as shape} s-flakes pid->p-flakes]
   (go-try
     (let [sid (flake/s (first s-flakes))]
       (log/debug "validate-shape" sid shape )
       (loop [[{:keys [path rhs-property qualified-value-shape] :as p-shape} & r] property
              q-shapes             []
-             validated-properties #{}
+             validated-properties validated-properties
```

Comment:
Are we double-checking constraints that pass advanced validation? I've been wondering if we could use something like`validated-properties` to avoid double-checking.

Reply:
We are not double-checking anything anymore, we actually segregate each shacl shape's property shapes into two buckets: advanced validation and regular validation.

We only use this `validated-properties` set to enable checking the `sh:closed` constraint.